### ISSUE TITLE
Wrong LA intersects test result

### ISSUE THREAD

**Issue by bjornharrtell** [bjornharrtell/jsts#202]
When doing intersects between `LINESTRING(240 190, 120 120)` and `POLYGON((110 240, 50 80, 240 70, 110 240))` something goes wrong and wrong output is generated. Expected output is `LINESTRING(177 153, 120 120)` but I get `LINESTRING(240 190,120 120)`.

Current tracing efforts is in SimpleMCSweepLineIntersector method processOverlaps:

```
# processOverlaps calls in JSTS:
processOverlaps start: 0 end: 1
processOverlaps start: 2 end: 3
processOverlaps start: 4 end: 5
processOverlaps start: 0 end: 1
processOverlaps start: 2 end: 3
processOverlaps start: 4 end: 5
processOverlaps start: 6 end: 7

# processOverlaps calls in JTS
processOverlaps start: 0 end: 1
processOverlaps start: 0 end: 3
processOverlaps start: 1 end: 4
processOverlaps start: 2 end: 5
processOverlaps start: 0 end: 3
processOverlaps start: 1 end: 6
processOverlaps start: 2 end: 7
processOverlaps start: 4 end: 5

**Comment by bjornharrtell** [comment]
It seems this is at least partially caused by the overloaded add method in SimpleMCSweepLineIntersector and that the original source relies on beeing able to call it with null values which cannot be handled by java2estree, see https://github.com/bjornharrtell/java2estree/issues/25.

**Comment by bjornharrtell** [comment]
The SweepLineEvent is one dangerous beast, I think I need to rewrite it...

**Comment by bjornharrtell** [comment]
After rewrite I'm still getting wrong output:

```
processOverlaps start: 0 end: 1
processOverlaps start: 0 end: 1
processOverlaps start: 2 end: 3
processOverlaps start: 4 end: 5
processOverlaps start: 0 end: 1
processOverlaps start: 2 end: 3
processOverlaps start: 4 end: 5
processOverlaps start: 6 end: 7

**Comment by bjornharrtell** [comment]
For yet unknown reasons the SweepLineEvent aren't correctly referenced when assigned new deleteEventIndex at SimpleMCSweepLineIntersector.prepareEvents.

**Comment by bjornharrtell** [comment]
Fixed by ongoing work at #178.
